src/Parser/Druid/Restoration/Modules/Talents/Photosynthesis.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+
+import StatisticBox from 'Main/StatisticBox';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatNumber, formatPercentage } from 'common/format';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+const PHOTOSYNTHESIS_REJUV_INCREASE = 0.3;
+
+class Photosynthesis extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  rejuvenationIncrease = 0;
+  lifebloomIncrease = 0;
+
+  lastRealBloomTimestamp = null;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTalent(SPELLS.PHOTOSYNTHESIS_TALENT.id);
+  }
+
+  on_byPlayer_refreshbuff(event) {
+    const spellId = event.ability.guid;
+
+    if (spellId === SPELLS.LIFEBLOOM_HOT_HEAL.id) {
+      this.lastRealBloomTimestamp = event.timestamp;
+    }
+  }
+
+  on_byPlayer_removebuff(event){
+    const spellId = event.ability.guid;
+    if(spellId !== SPELLS.LIFEBLOOM_HOT_HEAL.id) {
+      return;
+    }
+    this.lastRealBloomTimestamp = event.timestamp;
+  }
+
+  on_byPlayer_heal(event) {
+    const spellId = event.ability.guid;
+    const amount = event.amount + (event.absorbed || 0);
+
+    if(spellId === SPELLS.REJUVENATION.id && this.combatants.selected.hasBuff(SPELLS.LIFEBLOOM_HOT_HEAL.id)) {
```

Comment:
Remember to check if it is the player's own lifebloom and not another resto druid's.